addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/dataexchange/NukiHttpClient.java

```diff
@@ -0,0 +1,170 @@
+import org.eclipse.jetty.client.HttpResponseException;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.openhab.binding.nuki.NukiBindingConstants;
+import org.openhab.binding.nuki.dto.BridgeApiInfoDto;
+import org.openhab.binding.nuki.dto.BridgeApiLockActionDto;
+import org.openhab.binding.nuki.dto.BridgeApiLockStateDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link NukiHttpClient} class is responsible for getting data from the Nuki Bridge.
+ *
+ * @author Markus Katter - Initial contribution
+ */
+public class NukiHttpClient {
+
+    private final Logger logger = LoggerFactory.getLogger(NukiHttpClient.class);
+
+    private HttpClient httpClient;
+    private Configuration configuration;
+
+    public NukiHttpClient(Configuration configuration) {
+        logger.trace("Instantiating NukiHttpClient({})", configuration);
+        this.configuration = configuration;
+        this.httpClient = new HttpClient();
+        long connectTimeout = NukiBindingConstants.CLIENT_CONNECT_TIMEOUT;
+        httpClient.setConnectTimeout(connectTimeout);
+    }
+
+    public void startClient() {
+        logger.trace("Starting HttpClient");
+        try {
+            httpClient.start();
+            logger.trace("Started httpClient[{}]", httpClient);
+        } catch (Exception e) {
+            logger.error("Could not start NukiHttpClient! ERROR: {}", e.getMessage(), e);
+        }
+    }
+
+    public void stopClient() {
+        logger.trace("Stopping HttpClient");
+        try {
+            if (httpClient.isStarted()) {
+                httpClient.stop();
+                logger.trace("Stopped NukiHttpClient[{}]", httpClient);
+            }
+        } catch (Exception e) {
+            logger.error("Could not stop NukiHttpClient! ERROR: {}", e.getMessage(), e);
+        }
+    }
+
+    private String prepareUri(String uriTemplate, String... additionalArguments) {
+        String configIp = (String) configuration.get(NukiBindingConstants.CONFIG_IP);
+        BigDecimal configPort = (BigDecimal) configuration.get(NukiBindingConstants.CONFIG_PORT);
+        String configApiToken = (String) configuration.get(NukiBindingConstants.CONFIG_API_TOKEN);
+        ArrayList<String> parameters = new ArrayList<>();
+        parameters.add(configIp);
+        parameters.add(configPort.toString());
+        parameters.add(configApiToken);
+        if (additionalArguments != null) {
+            for (String argument : additionalArguments) {
+                parameters.add(argument);
+            }
+        }
+        String uri = String.format(uriTemplate, parameters.toArray());
+        logger.trace("uri[{}]", uri);
+        return uri;
+    }
+
+    private ContentResponse executeRequest(String uri)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        startClient();
```

Comment:
Well, I first implemented it like you suggested but I ran into "concurrent execution exception" (something like this) during openHAB startup / binding initialization.
Also, because every service method (e.g. getBridgeInfo(), etc.) just use executeRequest() just once it should be ok. If a service method would need to call more than one endpoint, I totally agree to start the client just one.